### PR TITLE
ENH: iPython extension for interactive object representation

### PR DESCRIPTION
**Pull request by tupui**
I propose to add one (or more ) extension to enable rich object representations in iPython.

This was discussed on the mailing list here:
https://mail.python.org/pipermail/scipy-dev/2021-May/024772.html

The current PR adds an extension which adds a display method rendering a PNG when calling a frozen distribution. It just shows the PDF, but I could add the CDF as well, or something else depending on wishes. It can be used as followed:

```python
%load_ext scipy.ipython.distributions
from scipy.stats import norm
norm(loc=3, scale=2)
```

Unknown

The same logic could be used in more modules. I have a few ideas for QMC to represent a sample for instance. Also an optimization result could show the search. Or just adding some latex equations in some places could make things prettier. Etc.

For now I created a new module `scipy.ipython` to store these in order to keep the current classes free from these extensions. I am also not sure how to test these. If we continue, I would add some doc somewhere (and I could include this in some tutorial or make a new one).

cc @rkern @Carreau

### CODE REVIEW

**Comment by rkern** [comment]
I recommend starting off in a separate repo as a separate package. Then we can play with these with currently-installed versions of scipy. You can iterate outside of the scipy release cycle (and build process). It will just be more useful to everyone if this starts out separate.

File: scipy/ipython/distributions.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ax.set_title(title)
+    ax.set_ylabel(r"$f$")
+    ax.set_xlabel(r"$x$")
+
+    data = print_figure(fig, "png")
+
+    # We MUST close the figure, otherwise IPython's display machinery
+    # will pick it up and send it as output, resulting in a double display
+    plt.close(fig)
+    return data
```
[the remark this answers is not in the document]
Comment (tupui) [reply scipy/ipython/distributions.py:35]: True, I will see to add something. I have a few ideas for doing this 😉

**Comment by tupui** [comment]
@rkern @Carreau I created this repository: https://github.com/tupui/scipy-magic. I would be happy to get your view on this. Feel free to open issues (I opened a general one to start with), and I can also give you rights.

**Comment by tupui** [comment]
I am not sure of closing this now as on the mail list it seems like people would like to have more information from the object (which might just be textual information for now). Might be another issue to open 🤷‍♂️

**Comment by mdhaber** [comment]
I like the idea and the name : ) Sounds good as a separate repo for now. It is a lot more flexible that way, which I think will be a good thing as ideas develop.

**Comment by tupui** [comment]
Thanks @mdhaber. Feel free to drop by and if you're interested I would be happy to add you as a contributor over there 😃

**Pull request closed without merging**